src/commands/showReferrer.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+import { execAsync } from '../support/execAsync';
+import { window } from 'vscode';
+import * as os from 'os';
+import * as path from 'path';
+import { access } from 'fs/promises'; //this import accepts one argument for calling access
+//TODO: Error Wrapping
+// check oras exist on user computer
+async function getOrasPath() {
+    const isWindows = os.platform() === 'win32'; //identifying if its a windows platform using os.platform, if it is then isWindows will be true
+    const homedir = os.homedir(); //getting the host computer home directory
+    const orasExecutable = isWindows ? 'oras.exe' : 'oras'; //switches use of executable files. if true use oras.exe and if false use oras
+    const dirOras = path.join(homedir, 'bin', 'oras', orasExecutable); //returns the path to oras
+
+    try {
+        await access(dirOras); // Check if the file exists, this will throw an error if the file doesn't exist
+        return dirOras; // If the file exists, return the dirOras path
```

Comment:
Same as above. Comments are a good tool while developing, especially while figuring out things that are not very well understood, however there is such a thing as over commenting which then reduces readability of code by later users. Personally, I try to follow the following:

- Comments should not duplicate the code.
- Variable or function names should be self-explanatory. Comments should enhance information on usage rather than function.
- Comments should typically be present at the function level rather than inline, except when calling external code that cannot be commented by you
- Comments should clarify otherwise unclear or unidiomatic code. They can also be used to group logical sets of actions in the code to enhance readability when separate functions are not justified.